### PR TITLE
feat: support vls128 for ros2

### PR DESCRIPTION
**Pull request by wep21**
ros2 port for #370, #382, #413

### CODE REVIEW

**Comment by JWhitleyWork** [comment]
Has this been tested on real hardware?

**Comment by wep21** [comment]
Not yet. I will test on real sensor.

**Comment by Mtestor** [comment]
I have tested and it doesn't work.

File: velodyne_pointcloud/params/VLS128.yaml [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+  focal_slope: 0.0, laser_id: 125, rot_correction: 0.04768239516448509, vert_correction: 0.0651007810993885,
+  vert_offset_correction: 0.0}
+- {dist_correction: 0.0, dist_correction_x: 0.0, dist_correction_y: 0.0, focal_distance: 0.0,
+  focal_slope: 0.0, laser_id: 126, rot_correction: 0.07937757438070212, vert_correction: -0.08656833089891874,
+  vert_offset_correction: 0.0}
+- {dist_correction: 0.0, dist_correction_x: 0.0, dist_correction_y: 0.0, focal_distance: 0.0,
+  focal_slope: 0.0, laser_id: 127, rot_correction: 0.1108982206717197, vert_correction: 0.007504915783575617,
+  vert_offset_correction: 0.0}
+num_lasers: 128
+distance_resolution: 0.007
```
Comment (Mtestor) [inline velodyne_pointcloud/params/VLS128.yaml:387]: Its 0.004 in the manual user.

File: velodyne_pointcloud/src/lib/rawdata.cpp [hunk 2]
```diff
@@ -399,6 +448,156 @@ void RawData::unpack(
...
+  uint16_t azimuth_next{};
+  uint16_t azimuth_corrected{};
+  float x_coord, y_coord, z_coord;
+  float distance;
+  const raw_packet * raw = (const raw_packet *) &pkt.data[0];
+  union two_bytes tmp;
+
+  float cos_vert_angle, sin_vert_angle, cos_rot_correction, sin_rot_correction;
+  float cos_rot_angle, sin_rot_angle;
+  float xy_distance;
```
Comment (Mtestor) [inline velodyne_pointcloud/src/lib/rawdata.cpp:472]: Won't it better to place the declaration where its used ?

**Review by Mtestor — CHANGES_REQUESTED** [review]

**Comment by wep21** [comment]
@Mtestor Thank you for testing. Could you create a PR to my branch?

**Comment by Mtestor** [comment]
sure

**Comment by Mtestor** [comment]
@wep21 your branch doesn't appear when i try to pull request.

**Comment by wep21** [comment]
@Mtestor You mean you cannot find my branch even when creating a PR from my [fork](https://github.com/wep21/velodyne/pulls)?

**Comment by Mtestor** [comment]
Thanks ! I have done a PR.

**Comment by JWhitleyWork** [comment]
Rebased on current `ros2` branch, which includes updated CI job.

**Review by JWhitleyWork — APPROVED** [review]
With rebase and some minor fixes, this looks good!

**Pull request merged**